native/service/governance/neo3_state_manager/neo3_state_manager.go

```diff
@@ -0,0 +1,158 @@
+	}
+
+	//check consensus signs
+	ok, err := node_manager.CheckConsensusSigns(native, APPROVE_REGISTER_STATE_VALIDATOR, utils.GetUint64Bytes(params.ID), params.Address)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRegisterStateValidator, CheckConsensusSigns error: %v", err)
+	}
+	if !ok {
+		return utils.BYTE_FALSE, nil
+	}
+
+	// put all the state validators in storage
+	err = putStateValidators(native, svListParam.StateValidators)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRegisterStateValidator, putStateValidators error: %v", err)
+	}
+
+	native.GetCacheDB().Delete(utils.ConcatKey(utils.Neo3StateManagerContractAddress, []byte(STATE_VALIDATOR_APPLY), utils.GetUint64Bytes(params.ID)))
+	native.AddNotify(
+		&event.NotifyEventInfo{
+			ContractAddress: utils.Neo3StateManagerContractAddress,
+			States:          []interface{}{"ApproveRegisterStateValidator", params.ID},
+		})
+	return utils.BYTE_TRUE, nil
+}
+
+func RemoveStateValidator(native *native.NativeService) ([]byte, error) {
+	params := new(StateValidatorListParam)
+	if err := params.Deserialization(common.NewZeroCopySource(native.GetInput())); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("RemoveStateValidator, contract params deserialize error: %v", err)
+	}
+	//check witness
+	if err := utils.ValidateOwner(native, params.Address); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("RemoveStateValidator, checkWitness: %s, error: %v", params.Address.ToBase58(), err)
+	}
+	err := putStateValidatorRemove(native, params)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("RemoveStateValidator, putStateValidatorRemove error: %v", err)
+	}
+	return utils.BYTE_TRUE, nil
+}
+
+func ApproveRemoveStateValidator(native *native.NativeService) ([]byte, error) {
+	params := new(ApproveStateValidatorParam)
+	if err := params.Deserialization(common.NewZeroCopySource(native.GetInput())); err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRemoveStateValidator, contract params deserialize error: %v", err)
+	}
+
+	//check witness
+	err := utils.ValidateOwner(native, params.Address)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRemoveStateValidator, checkWitness error: %v", err)
+	}
+
+	svListParam, err := getStateValidatorRemove(native, params.ID)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRemoveStateValidator, getStateValidatorRemove error: %v", err)
+	}
+
+	//check consensus signs
+	ok, err := node_manager.CheckConsensusSigns(native, APPROVE_REMOVE_STATE_VALIDATOR, utils.GetUint64Bytes(params.ID), params.Address)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRemoveStateValidator, CheckConsensusSigns error: %v", err)
+	}
+	if !ok {
+		return utils.BYTE_FALSE, nil
+	}
+
+	// remove svs
+	err = removeStateValidators(native, svListParam.StateValidators)
+	if err != nil {
+		return utils.BYTE_FALSE, fmt.Errorf("ApproveRemoveStateValidator, removeStateValidators error: %v", err)
+	}
+
```

Comment:
After finished remove validators, why not delete the removeid from db?

Reply:
udpated